tests/phpunit/tests/blocks/register.php

```diff
@@ -372,6 +372,56 @@ public function test_success_register_block_style_handle() {
 			wp_normalize_path( realpath( DIR_TESTDATA . '/blocks/notice/block.css' ) ),
 			wp_normalize_path( wp_styles()->get_data( 'unit-tests-test-block-style', 'path' ) )
 		);
+
+	}
+
+	/**
+	 * Tests that register_block_style_handle() loads RTL stylesheets when an RTL locale is set.
+	 *
+	 * @ticket 56797
+	 *
+	 * @covers ::register_block_style_handle
+	 */
+	public function test_register_block_style_handle_should_load_rtl_stylesheets_for_rtl_text_direction() {
+		global $wp_locale;
+
+		$metadata = array(
+			'file'  => DIR_TESTDATA . '/blocks/notice/block.json',
+			'name'  => 'unit-tests/test-block-rtl',
+			'style' => 'file:./block.css',
+		);
+
+		$orig_text_dir             = $wp_locale->text_direction;
+		$wp_locale->text_direction = 'rtl';
+		$handle                    = register_block_style_handle( $metadata, 'style' );
+		$extra_rtl                 = wp_styles()->get_data( 'unit-tests-test-block-rtl-style', 'rtl' );
+		$extra_suffix              = wp_styles()->get_data( 'unit-tests-test-block-rtl-style', 'suffix' );
+		$extra_path                = wp_normalize_path( wp_styles()->get_data( 'unit-tests-test-block-rtl-style', 'path' ) );
+		$wp_locale->text_direction = $orig_text_dir;
+
+		$this->assertSame(
+			'unit-tests-test-block-rtl-style',
+			$handle,
+			'The handle does not match the expected handle.'
```

Comment:
Small inconsistency: the other feedback messages use past tense, and this one use present :)

Reply:
You know, every time I write an assertion message, I start to wonder if I've switched tense at some stage. And now I finally have, or someone has finally caught it! 😆

Reply:
Fixed in [e43ec26](https://github.com/WordPress/wordpress-develop/pull/4150/commits/e43ec26baf5936b8bf66b6fa33364cc905c01c5d).